src/Nethermind/Nethermind.Evm.Benchmark/IlEvmBenchmarksBase.cs

```diff
@@ -0,0 +1,620 @@
+                        .Done;
+
+        static byte[] isPrimeBytecode2(byte[] argBytes) => Prepare.EvmCode
+                        .JUMPDEST()
+                        .PUSHx([0])
+                        .POP()
+                        .PUSHx(argBytes)
+                        .COMMENT("Store variable(n) in Memory")
+                        .MSTORE(0)
+                        .COMMENT("Store Indexer(i) in Memory")
+                        .PushData(2)
+                        .MSTORE(32)
+                        .COMMENT("We mark this place as a GOTO section")
+                        .JUMPDEST()
+                        .COMMENT("We check if i * i < n")
+                        .MLOAD(32)
+                        .DUPx(1)
+                        .MUL()
+                        .MLOAD(0)
+                        .LT()
+                        .PushData(4 + 47 + argBytes.Length)
+                        .JUMPI()
+                        .COMMENT("We check if n % i == 0")
+                        .MLOAD(32)
+                        .MLOAD(0)
+                        .MOD()
+                        .ISZERO()
+                        .DUPx(1)
+                        .COMMENT("if 0 we jump to the end")
+                        .PushData(4 + 51 + argBytes.Length)
+                        .JUMPI()
+                        .POP()
+                        .COMMENT("increment Indexer(i)")
+                        .MLOAD(32)
+                        .ADD(1)
+                        .MSTORE(32)
+                        .COMMENT("Loop back to top of conditional loop")
+                        .PushData(4 + 9 + argBytes.Length)
+                        .JUMP()
+                        .COMMENT("return 0")
+                        .JUMPDEST()
+                        .PushData(0)
+                        .STOP()
+                        .JUMPDEST()
+                        .Done;
+    }
+
+
+    public abstract class IlEvmBenchmark : IlEvmBenchmarkBase
+    {
+        protected virtual byte[] GenerateBytecode(byte[] argBytes) => Array.Empty<byte>();
+
+        public virtual ulong Value { get; set; } = 1000UL;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+
+            Initialize();
+            if (TransactionSet.Length == 0)
+            {
+                byte[] bytes = new byte[32];
+                new UInt256(Value).ToBigEndian(bytes);
+                var argBytes = bytes.WithoutLeadingZeros().ToArray();
+                SetCode(GenerateBytecode(argBytes));
+                return;
+            }
+
+
+            InsertCode(bytecode, ContractAddress);
+            BuildBlock();
+
+            IlAnalyzer.StartPrecompilerBackgroundThread(vmConfigOptimizing ?? new VMConfig(), NullLogger.Instance);
+
+            SeedAccounts();
+            snapshot = _stateProvider!.TakeSnapshot();
+        }
+
+        [IterationSetup]
+        public void Setup()
+        {
+            if (TransactionSet.Length == 0)
```

Comment:
We should split it. It's either a single execution or a transaction. Should be kept separate.